src/ripple/nodestore/impl/Shard.h

```diff
@@ -46,9 +46,8 @@ using NCache = KeyCache<uint256>;
 class Shard
 {
 public:
-    Shard(std::uint32_t index, int cacheSz,
-        PCache::clock_type::rep cacheAge,
-        beast::Journal& j);
+    Shard(DatabaseShard const& db, std::uint32_t index, int cacheSz,
```

Comment:
On non-unity builds, `DatabaseShard` is not defined here. A pre-declaration around line 40 appears to resolve the issue.

Reply:
@ximinez Thanks, pushed fix.